Batteries/Data/Array/Lemmas.lean

```diff
@@ -126,3 +126,8 @@ theorem size_shrink_loop (a : Array α) (n) : (shrink.loop n a).size = a.size -
 theorem size_shrink (a : Array α) (n) : (a.shrink n).size = min a.size n := by
   simp [shrink, size_shrink_loop]
   omega
+
+/-! ### map -/
+
+unseal mapM.map in
+theorem map_empty (f : α → β) : map f empty = empty := rfl
```

Comment:
Since `mapM` needs to be mentioned anyway, we might as well have the matching pair.
```suggestion
theorem mapM_empty [Monad m] (f : α → m β) : mapM f #[] = pure #[] := by
  rw [mapM, mapM.map]; rfl

@[simp] theorem map_empty (f : α → β) : map f #[] = #[] := mapM_empty ..
```

Reply:
Happy to accept this. At this point, I just hope these PRs will get merged some day.